src/main/groovy/com/github/winplay02/gitcraft/manifest/skyrising/SkyrisingMetadataProvider.java

```diff
@@ -0,0 +1,119 @@
+package com.github.winplay02.gitcraft.manifest.skyrising;
+
+import com.github.winplay02.gitcraft.manifest.BaseMetadataProvider;
+import com.github.winplay02.gitcraft.manifest.ManifestSource;
+import com.github.winplay02.gitcraft.meta.VersionDetails;
+import com.github.winplay02.gitcraft.meta.VersionInfo;
+import com.github.winplay02.gitcraft.types.OrderedVersion;
+import com.github.winplay02.gitcraft.util.MiscHelper;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+
+public class SkyrisingMetadataProvider extends BaseMetadataProvider<SkyrisingManifest, SkyrisingManifest.VersionEntry> {
+
+	private final Map<String, VersionDetails> versionDetails = new HashMap<>();
+
+	public SkyrisingMetadataProvider() {
+		this.addManifestSource("https://skyrising.github.io/mc-versions/version_manifest.json", SkyrisingManifest.class);
+	}
+
+	@Override
+	public ManifestSource getSource() {
+		return ManifestSource.SKYRISING;
+	}
+
+	@Override
+	public String getName() {
+		return "Skyrising Version Metadata (https://skyrising.github.io/mc-versions/)";
+	}
+
+	@Override
+	public String getInternalName() {
+		return "skyrising";
+	}
+
+	@Override
+	protected OrderedVersion loadVersionFromManifest(SkyrisingManifest.VersionEntry manifestEntry, Path targetDir) throws IOException {
+		VersionInfo info = this.fetchVersionMetadata(manifestEntry.id(), manifestEntry.url(), manifestEntry.sha1(), targetDir.resolve("info"), "version info", VersionInfo.class);
```

Comment:
Since it is possible to fetch different manifests for the same version, I think they should all be cached. I tried to do that by including the sha1 hash in the cached name. What do you think about that?

Reply:
That sounds good. Its about the manifests listed in the details files, right? I think it's the best solution for those